source/extensions/filters/udp/udp_proxy/udp_proxy_filter.cc

```diff
@@ -274,6 +280,35 @@ UdpProxyFilter::ActiveSession::~ActiveSession() {
       ->resourceManager(Upstream::ResourcePriority::Default)
       .connections()
       .dec();
+
+  if (!cluster_.filter_.config_->accessLogs().empty()) {
+    fillStreamInfo();
+    for (const auto& access_log : cluster_.filter_.config_->accessLogs()) {
+      access_log->log(nullptr, nullptr, nullptr, udp_sess_stats_.value());
+    }
+  }
+}
+
+void UdpProxyFilter::ActiveSession::fillStreamInfo() {
+  ProtobufWkt::Struct stats_obj;
+  auto& fields_map = *stats_obj.mutable_fields();
+  fields_map["cluster_name"] = ValueUtil::stringValue(cluster_.cluster_.info()->name());
+  fields_map["bytes_sent"] =
+      ValueUtil::numberValue(session_stats_.downstream_sess_tx_bytes_.value());
+  fields_map["bytes_received"] =
+      ValueUtil::numberValue(session_stats_.downstream_sess_rx_bytes_.value());
+  fields_map["errors_sent"] =
+      ValueUtil::numberValue(session_stats_.downstream_sess_tx_errors_.value());
+  // TODO(giantcroc): Since the receiving errors are counted in onReceiveError, it seems that
+  // no way to distinguish the number of errors in each session.
+  fields_map["errors_received"] =
+      ValueUtil::numberValue(cluster_.filter_.config_->stats().downstream_sess_rx_errors_.value());
```

Comment:
Then I would either remove this or make this very clear in the docs that this is global to the listener.

Reply:
> Then I would either remove this or make this very clear in the docs that this is global to the listener.

OK, I have moved this comment to dynamic metadata in usage.